segmentstore/server/src/main/java/io/pravega/segmentstore/server/logs/operations/MergeSegmentOperation.java

```diff
@@ -113,17 +126,28 @@ public long getLength() {
     @Override
     public String toString() {
         return String.format(
-                "%s, SourceSegmentId = %d, Length = %s, MergeOffset = %s",
+                "%s, SourceSegmentId = %d, Length = %s, MergeOffset = %s, Attributes = %d",
                 super.toString(),
                 getSourceSegmentId(),
                 toString(getLength(), -1),
-                toString(getStreamSegmentOffset(), -1));
+                toString(getStreamSegmentOffset(), -1),
+                this.attributeUpdates.size());
+    }
+
+    //endregion
+
+    //region AttributeUpdaterOperation implementation
+
+    @Override
+    public AttributeUpdateCollection getAttributeUpdates() {
+        return AttributeUpdateCollection.from(attributeUpdates);
```

Comment:
Why do you make a collection out of the collection? It's supposed to be immutable in the first place.

Reply:
My fault, fixed (actually, I fixed that at another point as well).